src/cascadia/TerminalSettingsModel/Profile.cpp

```diff
@@ -151,22 +157,22 @@ bool Profile::ShouldBeLayered(const Json::Value& json) const
 
     // For profiles with a `source`, also check the `source` property.
     bool sourceMatches = false;
-    if (!_Source.empty())
+    if (_Source && !_Source.value().empty())
     {
         if (otherHadSource)
         {
             // If we have a source and the other has a source, compare them!
-            sourceMatches = *otherSource == _Source;
+            sourceMatches = *otherSource == *_Source;
         }
         else
         {
             // Special case the legacy dynamic profiles here. In this case,
             // `this` is a dynamic profile with a source, and our _source is one
             // of the legacy DPG namespaces. We're looking to see if the other
             // json object has the same guid, but _no_ "source"
-            if (_Source == WslGeneratorNamespace ||
-                _Source == AzureGeneratorNamespace ||
-                _Source == PowershellCoreGeneratorNamespace)
+            if (*_Source == WslGeneratorNamespace ||
+                *_Source == AzureGeneratorNamespace ||
+                *_Source == PowershellCoreGeneratorNamespace)

```

Comment:
careful with source. source may need to be a "see-through" property. Gotta handle it appropriately.